### PR TITLE
make freeze work recursively

### PR DESCRIPTION
**Pull request by orionz**
Enhancing the JavaScript wrapper to respect the `{ freeze: true }` flag recursively and behave properly on `load()` and `clone()`.

### CODE REVIEW

File: crates/automerge-wasm/src/interop.rs [hunk 1]
```diff
@@ -370,20 +370,23 @@ impl Automerge {
         datatype: Datatype,
         heads: Option<&Vec<ChangeHash>>,
         meta: &JsValue,
+        freeze: bool,
```
Comment (alexjg) [inline crates/automerge-wasm/src/interop.rs:373]: If I understand correctly the purpose of this `freeze` argument is to track whether the object being patched (`nextDoc` in `progressDocument` should be `Object.freeze`ed after the patch is applied. Currently this only seems to be the case if the `Automerge` document was initialized with `freeze: true`, would it be possible then to just track the freeze state via a field on `automerge_wasm::Automerg` (e.g. `automerge_wasm::Automerge::freeze`)? That way we wouldn't need to feed the `freeze` argument through to every one of these functions.

File: wrappers/javascript/test/basic_test.ts [hunk 2]
```diff
@@ -21,6 +21,50 @@ describe('Automerge', () => {
...
+            let doc1 = Automerge.from({ sub: { object: true } })
+            assert(Automerge.isAutomerge(doc1))
+            assert(!Automerge.isAutomerge(doc1.sub))
+            assert(!Automerge.isAutomerge("String"))
+            assert(!Automerge.isAutomerge({ sub: { object: true }}))
+            assert(!Automerge.isAutomerge(undefined))
+        })
+
+        it('it should recursively freeze the document if requested', () => {
+            let doc1 = Automerge.init({ freeze: true } )
```
Comment (alexjg) [inline wrappers/javascript/test/basic_test.ts:34]: This feels like an API that will be surprising to javascript programmers, is it possible to hook into `Object.freeze` at all so folks can just do:

```
let doc1 = Automerge.init()
Object.freeze(doc1)
```

File: crates/automerge-wasm/src/interop.rs (hunk 1 above)
Comment (orionz) [reply crates/automerge-wasm/src/interop.rs]: Ok - good call - I added an `enableFreeze()` method which I don't love but it shadows the `enabledPatches()` which is also a stand-in for a better API later.  We can fix it up at a later date.

File: wrappers/javascript/test/basic_test.ts (hunk 2 above)
Comment (orionz) [reply wrappers/javascript/test/basic_test.ts]: Ok.  Fixed.

**Comment by orionz** [comment]
These failed test seem to be coming from main and not this branch.  Going to merge but we need to get those errors fixed asap.

**Pull request merged**